apis/topology/v1beta1/topology_types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// TopologySpec defines the desired state of Topology
+type TopologySpec struct {
+	// +kubebuilder:validation:Optional
+	// APITopologySpreadConstraint exposes topologySpreadConstraint that are
+	// applied to the StatefulSet
+	TopologySpreadConstraint *[]corev1.TopologySpreadConstraint `json:"topologySpreadConstraint,omitempty"`
+
+	// APIAffinity exposes PodAffinity and PodAntiaffinity overrides that are applied
+	// to StatefulSet/Deployments
+	// +optional
+	APIAffinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	//TODO: We could add NodeSelector here as it belongs to the same APIGroup
+}
+
+// TopologyStatus defines the observed state of Topology
+type TopologyStatus struct {
+
+	// ObservedGeneration - the most recent generation observed for this
```

Comment:
nit: comment refers to service, ContainerImage etc... which don't apply here, top-level CR does not manage this.

Reply:
Done